CMakeLists.txt

```diff
@@ -325,12 +327,10 @@ find_package(Boost 1.49
 
 link_directories(${Boost_LIBRARY_DIRS})
 
-# In Visual Studio, automatic linking is performed, so we don't need to worry
-# about it.  Clear the list of libraries to link against and let Visual Studio
-# handle it.
+# Disable automatic linking in Visual Studio.
 if (MSVC)
   link_directories(${Boost_LIBRARY_DIRS})
-  set(Boost_LIBRARIES "")
+  add_definitions(-DBOOST_ALL_NO_LIB)
```

Comment:
Do you have to define ``BOOST_ALL_DYN_LINK`` as well?

Reply:
I'm not 100% sure.  My impression was that by defining `BOOST_ALL_NO_LIB`, then Visual Studio would not automatically link the libraries, but CMake would configure the project to manually link the Boost libraries.  If you know more than me on this one, I'm happy to add `BOOST_ALL_DYN_LINK` also.

Reply:
I just noticed the build failed and the only difference I can see between #1524 and this PR is the ``BOOST_ALL_DYN_LINK`` option. Not sure this will fix the windows build, maybe we have to adapt the build steps as well.

Reply:
Oh, I see, I didn't realize the build was failing.  I'm not sure why I didn't notice that...

I'll try some debugging and see where it goes. :)